client/reader/sidebar/index.jsx

```diff
@@ -254,22 +235,51 @@ const ReaderSidebar = React.createClass( {
 					onTagExists={ this.highlightNewTag }
 					currentTag={ this.state.currentTag } />
 
-				</SidebarRegion>
+			</SidebarRegion>
 
-				{ this.renderAppPromo() }
+			{ this.props.shouldRenderAppPromo &&
+				<div className="reader-sidebar__AppPromo">
+					<AppPromo location="reader" locale={ userUtils.getLocaleSlug() } />
+				</div>
+			}
 
 				<SidebarFooter />
 			</Sidebar>
 		);
 	}
 } );
 
+ReaderSidebar.defaultProps = {
+	translate: identity
+};
+
+export const shouldRenderAppPromo = ( options = { } ) => {
+	const {
+		isDesktopPromoDisabled = store.get( 'desktop_promo_disabled' ),
+		isViewportMobile = viewport.isMobile(),
+		isUserLocaleEnglish = 'en' === userUtils.getLocaleSlug(),
+		isDesktopPromoConfiguredToRun = config.isEnabled( 'desktop-promo' ),
+		isUserDesktopAppUser = userSettings.getSetting( 'is_desktop_app_user' ),
+		isUserOnChromeOS = /\bCrOS\b/.test( navigator.userAgent )
```

Comment:
Strange as it'll look in this case, the guidelines are such that even acronyms should be camelCased ([reference](https://github.com/Automattic/wp-calypso/blob/master/docs/coding-guidelines/javascript.md#naming-conventions)):

```js
isUserOnChromeOs
```